Documentation/cilium_operator.rst

```diff
@@ -0,0 +1,212 @@
+
+.. parsed-literal::
+
+    helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set operator.replicas=3
+
+.. code:: bash
+
+    $ kubectl get deployment cilium-operator -n kube-system
+    NAME              READY   UP-TO-DATE   AVAILABLE   AGE
+    cilium-operator   3/3     3            3           46s
+
+The operator is an integral part of Cilium installations in Kubernetes
+environments and is tasked to perform the following operations:
+
+CRD Registration
+~~~~~~~~~~~~~~~~
+
+The default behavior of cilium-operator is to register the CRDs used by
+Cilium. The following custom resources are registered by cilium-operator:
+
+-  CiliumNetworkPolicy
+-  CiliumClusterwideNetworkPolicy
+-  CiliumEndpoint
+-  CiliumNode
+-  CiliumExternalWorkload
+-  CiliumIdentity
+-  CiliumLocalRedirectPolicy
+
+IPAM
+~~~~
+
+Cilium-Operator is responsible for IP address management when running in
+the following modes:
+
+-  :ref:`ipam_azure`
+-  :ref:`ipam_eni`
+-  :ref:`ipam_crd_cluster_pool`
+
+When running in IPAM mode :ref:`k8s_hostscope` the allocation CIDRs used by
+``cilium-agent`` is derived from the fields ``podCIDR`` and ``podCIDRs``
+populated by Kubernetes in the Kubernetes ``Node`` resource.
+
+For :ref:`concepts_ipam_crd` IPAM allocation mode, it is the job of custom
+operator, such as: "cilium-operator-aws", "cilium-operator-azure" to populate
+the required information about CIDRs in the ``CiliumNode`` resource.
+
+For more information on IPAM visit :ref:`address_management`.
+
+KVStore operations
+~~~~~~~~~~~~~~~~~~
+
+These operations are performed only when KVStore is enabled for the
+operator. In addition KVStore operations are only required when
+``cilium-operator`` is running with any of the below options:
+
+-  ``--synchronize-k8s-services``
+-  ``--synchronize-k8s-nodes``
+-  ``--identity-allocation-mode=kvstore``
+
+K8s Services synchronization
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cilium operator performs the job of synchronizing Kubernetes services to
+external KVstore configured for the operator if running with
+``--synchronize-k8s-services`` flag.
+
+The operator performs this operation only for shared services(services
+that have ``io.cilium/shared-service`` annotation set to true). This is
+meaningful when running Cilium to setup a clustermesh.
+
+K8s Nodes synchronization
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Similar to K8s services, Cilium Operator also synchronize Kubernetes nodes
+information to the shared KVstore.
+
+When a ``Node`` object is deleted it is not possible to reliably cleanup
+the corresponding ``CiliumNode`` object from the agent itself. The operator
+holds the responsibility to garbage collect orphaned ``CiliumNodes``.
+
+CNP/CCNP node status GC
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Similar to the effect of ``Node`` object deletion on ``CiliumNode``,
+Cilium Operator cannot remove the status corresponding to a node in a
+CNP/CCNP object. This operation of node status garbage collection from
```

Comment:
Maybe expand the acronyms once? You use them elsewhere, but either in a list for CRD registration, or later in the doc. Maybe something like:
```suggestion
CiliumNetworkPolicy (CNP) or CiliumClusterwideNetworkPolicy (CCNP) object. This operation of node status garbage collection from
```